autoPyTorch/pipeline/components/training/trainer/utils.py

```diff
@@ -0,0 +1,187 @@
+import re
```

Comment:
Is this just one of the choices of optimizers?
I need to understand each component one by one.